src/main/java/cloud/eppo/EppoClient.java

```diff
@@ -412,9 +409,71 @@ private JsonNode parseJsonString(String jsonString) {
     }
   }
 
+  public BanditResult getBanditAction(
+      String flagKey,
+      String subjectKey,
+      DiscriminableAttributes subjectAttributes,
+      Actions actions,
+      String defaultValue) {
+    BanditResult result = new BanditResult(defaultValue, null);
+    try {
+      String assignedVariation =
+          getStringAssignment(
+              flagKey, subjectKey, subjectAttributes.getAllAttributes(), defaultValue);
+
+      // Update result to reflect that we've been assigned a variation
+      result = new BanditResult(assignedVariation, null);
+
+      String banditKey = configurationStore.banditKeyForVariation(flagKey, assignedVariation);
+      if (banditKey != null && !actions.isEmpty()) {
+        BanditParameters banditParameters = configurationStore.getBanditParameters(banditKey);
+        BanditEvaluationResult banditResult =
+            BanditEvaluator.evaluateBandit(
+                flagKey, subjectKey, subjectAttributes, actions, banditParameters.getModelData());
+
+        // Update result to reflect that we've been assigned an action
+        result = new BanditResult(assignedVariation, banditResult.getActionKey());
```

Comment:
@typotter we now set the action part of the result as soon as we get it, and surround bandit logging with a `try`-`catch` like we do assignment logging.

Non-fatal logging error tests were added as well.